lib/cartopy/util.py

```diff
@@ -99,3 +101,298 @@ def add_cyclic_point(data, coord=None, axis=-1):
+    >>> print(cyclic_lon2d)
+    [[  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]]
+
+    Adding a cyclic point to a data array and an associated 2-dimensional
+    coordinate and a second raw variable.
+    >>> lons = np.arange(0, 360, 60)
+    >>> lats = np.arange(-90, 90, 180/5)
+    >>> lon2d, lat2d = np.meshgrid(lons, lats)
+    >>> cyclic_data, cyclic_lon2d, cyclic_lat2d = add_cyclic(
+    ...     data, coord=lon2d, rowcoord=lat2d)
+    >>> print(cyclic_data)  # doctest: +NORMALIZE_WHITESPACE
+    [[ 0. 1. 2. 3. 4. 5. 0.]
+     [ 0. 1. 2. 3. 4. 5. 0.]
+     [ 0. 1. 2. 3. 4. 5. 0.]
+     [ 0. 1. 2. 3. 4. 5. 0.]
+     [ 0. 1. 2. 3. 4. 5. 0.]]
+    >>> print(cyclic_lon2d)
+    [[  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]
+     [  0  60 120 180 240 300 360]]
+    >>> print(cyclic_lat2d)
+    [[-90. -90. -90. -90. -90. -90. -90.]
+     [-54. -54. -54. -54. -54. -54. -54.]
+     [-18. -18. -18. -18. -18. -18. -18.]
+     [ 18.  18.  18.  18.  18.  18.  18.]
+     [ 54.  54.  54.  54.  54.  54.  54.]]
+
+    Not adding a cyclic point if cyclic point detected in coord.
+    >>> lons = np.arange(0, 361, 72)
+    >>> lats = np.arange(-90, 90, 180/5)
+    >>> lon2d, lat2d = np.meshgrid(lons, lats)
+    >>> cyclic_data, cyclic_lon2d, cyclic_lat2d = add_cyclic(
+    ...     data, coord=lon2d, rowcoord=lat2d)
+    >>> print(cyclic_data)  # doctest: +NORMALIZE_WHITESPACE
+    [[ 0. 1. 2. 3. 4. 5.]
+     [ 0. 1. 2. 3. 4. 5.]
+     [ 0. 1. 2. 3. 4. 5.]
+     [ 0. 1. 2. 3. 4. 5.]
+     [ 0. 1. 2. 3. 4. 5.]]
+    >>> print(cyclic_lon2d)
+    [[  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]
+     [  0  72 144 216 288 360]]
+    >>> print(cyclic_lat2d)
+    [[-90. -90. -90. -90. -90. -90.]
+     [-54. -54. -54. -54. -54. -54.]
+     [-18. -18. -18. -18. -18. -18.]
+     [ 18.  18.  18.  18.  18.  18.]
+     [ 54.  54.  54.  54.  54.  54.]]
+    """
+    if coord is None:
+        return _add_cyclic_data(data, axis=axis)
+    else:
+        if (coord.ndim > 2):
+            caxis = axis
+        else:
+            caxis = -1
+        if coord.shape[caxis] != data.shape[axis]:
+            estr = 'coord.shape[{}] does not match'.format(caxis)
+            estr += ' the size of the corresponding dimension of'
+            estr += ' the data array: coord.shape[{}] ='.format(caxis)
+            estr += ' {}, data.shape[{}] = {}.'.format(
+                coord.shape[caxis], axis, data.shape[axis])
+            raise ValueError(estr)
+        if _has_cyclic(coord, axis=caxis, cyclic=cyclic, prec=prec):
+            if rowcoord is None:
+                return data, coord
+            else:
+                return data, coord, rowcoord
+        else:
```

Comment:
You could remove this entire indented else block if you want since you've returned from within the if statement above.